prow/cluster/hook_deployment.yaml

```diff
@@ -30,10 +30,10 @@ spec:
       labels:
         app: hook
     spec:
-      terminationGracePeriodSeconds: 30
+      terminationGracePeriodSeconds: 180
```

Comment:
Yep. Right now it's removed from the lb, and I just assume it'll finish processing all events within the grace period, then it's killed. You're definitely right that it would be neater for it to decide when it's done and exit.